### PR TITLE
Qexperiment

### PR DESCRIPTION
**Pull request by ssmi1975**
implementing the idea based on the discussion in slack.
Implementation for `SympyExecutor` and `IBMQExecutor` have done. (Numpy is not yet done)

- qapply returns an object that depends on executer
  - SympyExecutor returns Sympy expression
  - IBMQExecutor returns qiskit "job" object
  - ClassicalSimulationExecutor : TBD
- qexperiment returns the list of (state in string, probability)

```
c = H(2)*H(1)*H(0)*Qubit('000')
result = qexperiment(c, executor=IBMQExecutor())
print(result)
# output: {'100': 132, '101': 118, '011': 107, '000': 149, '010': 116, '110': 142, '001': 120, '111': 140}
```

### CODE REVIEW

**Comment by ssmi1975** [comment]
TODO
- ~~change the name `qexperiment` to `measure_all`, following the equivalent sympy function.~~
- ~~Also implement `measure_all_oneshot`.~~

`measure_partial` may not be possible for all executors (it returns quantum states, so it's more like qapply())

**Comment by ssmi1975** [comment]
qexperiment is different from measure_all
- it's a shortcut method that executes qapply and measure_all. in real devices, we can't implement qapply so this shortcut may be important.
- it returns the occurrence of each state, not probability.

File: quantpy/sympy/executor/classical_simulation_executor.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def execute(self, circuit, **options):
         qobj = circuits_to_qobj(circuit, QasmSimulatorPy())
         json = qobj_to_dict(qobj)["experiments"][0]
         self.simulate(json)
-        return str(self.simulator)
+        return self.simulator.to_coefficients()
```
Comment (ssmi1975) [inline quantpy/sympy/executor/classical_simulation_executor.py:42]: is this change acceptable? @kyamaz 
previously this was a string like `0.5|0> + 0.5|1>`, but changed to `{'0': 0.5, '1': 0.5}`

I want to use this result in qexperiment.

Comment (kyamaz) [reply quantpy/sympy/executor/classical_simulation_executor.py:42]: I think it is pretty good to change output format of execute function in ClassicalSimulationExecutor. @ssmi1975

Comment (ssmi1975) [reply quantpy/sympy/executor/classical_simulation_executor.py:42]: thanks! I will keep this then.

**Comment by openql-org** [comment]
I have checked `pytest` on my local environment and Travis CI.
Thank you for your contribution. @ssmi1975

**Pull request merged**